src/Temporalio/Worker/TemporalWorkerOptions.cs

```diff
@@ -0,0 +1,131 @@
+
+namespace Temporalio.Worker
+{
+    /// <summary>
+    /// Options for a <see cref="TemporalWorker" />. <see cref="TaskQueue" /> and at least one
+    /// workflow or activity are required.
+    /// </summary>
+    public class TemporalWorkerOptions : ICloneable
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TemporalWorkerOptions"/> class.
+        /// </summary>
+        public TemporalWorkerOptions()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TemporalWorkerOptions"/> class.
+        /// </summary>
+        /// <param name="taskQueue">Task queue for the worker.</param>
+        public TemporalWorkerOptions(string taskQueue)
+        {
+            TaskQueue = taskQueue;
+        }
+
+        /// <summary>
+        /// Gets or sets the task queue for the worker.
+        /// </summary>
+        public string? TaskQueue { get; set; }
+
+        /// <summary>
+        /// Gets or sets the task queue for activities. Default is <see cref="Task.Factory" />.
+        /// </summary>
+        public TaskFactory ActivityTaskFactory { get; set; } = Task.Factory;
+
+#pragma warning disable CA2227 // Intentionally allow setting of this collection w/ options pattern
+        /// <summary>
+        /// Gets or sets the activity delegates.
+        /// </summary>
+        public IList<Delegate> Activities { get; set; } = new List<Delegate>();
+#pragma warning restore CA2227
+
+        /// <summary>
+        /// Gets or sets the interceptors. Note this automatically includes any
+        /// <see cref="Client.TemporalClientOptions.Interceptors" /> that implement
+        /// <see cref="Interceptors.IWorkerInterceptor" /> so those should not be specified here.
+        /// This set is chained after the set in the client options.
+        /// </summary>
+        public IEnumerable<Interceptors.IWorkerInterceptor>? Interceptors { get; set; }
+
+        /// <summary>
+        /// Gets or sets the build ID. This is a unique identifier for each "build" of the worker.
+        /// If unset, the default is
+        /// <c>Assembly.GetEntryAssembly().ManifestModule.ModuleVersionId</c>.
+        /// </summary>
+        public string? BuildID { get; set; }
+
+        /// <summary>
+        /// Gets or sets the identity for this worker. If unset, defaults to the client's identity.
+        /// </summary>
+        public string? Identity { get; set; }
+
+        /// <summary>
+        /// Gets or sets the maximum number of activities that will ever be given to this worker
+        /// concurrently.
+        /// </summary>
+        public int MaxConcurrentActivities { get; set; } = 100;
+
+        /// <summary>
+        /// Gets or sets the maximum number of concurrent poll activity task requests we will
+        /// perform at a time on this worker's task queue.
+        /// </summary>
+        public int MaxConcurrentActivityTaskPolls { get; set; } = 5;
```

Comment:
There's only going to be one with the current implementation, you might as we not expose this at all until this is fixed or we implement poller autoscaling which would make this obsolete.

Reply:
If there a general purpose issue where we're deprecating this or seeing it to one in other core SDKs?

Reply:
I didn't expose this in TS: https://typescript.temporal.io/api/interfaces/worker.WorkerOptions

Reply:
Just let Core do its thing.
Don't deprecate for now, I hope to revisit the core poller implementation soon.

Reply:
@bergundy You keep saying this, and I keep mentioning it's not actually true. It does, in fact, spin up multiple pollers. I just ran a test and looked at metrics:

```
temporal_num_pollers{namespace="default",poller_type="sticky_workflow_task",service_name="temporal-core-sdk",task_queue="workflow_load_YfXY5vL2"} 8
temporal_num_pollers{namespace="default",poller_type="workflow_task",service_name="temporal-core-sdk",task_queue="workflow_load_YfXY5vL2"} 2
```

There's something wrong with activity pollers always reporting 0 for the metric value right now, not sure that that's about, but I verified just throwing in a debug statement that activity pollers also got as high as 8 with a 10 value.

So, this _does in fact do something_

Reply:
IIUC, we may see more than 1 concurrent activity poller under certain circumstances but this isn't really intentional and can only happen if there are local activities or activity cancellations to be dispatched while there's an outstanding poll request.

The poll buffer guarantees that we do not poll unless a poll is requested:

https://github.com/temporalio/sdk-core/blob/17ceab7952fc67674b624183b62eae16ecd12824/core/src/pollers/poll_buffer.rs#L130

https://github.com/temporalio/sdk-core/blob/17ceab7952fc67674b624183b62eae16ecd12824/core/src/pollers/poll_buffer.rs#L82

We add a poll permit every time we reach this line:

https://github.com/temporalio/sdk-core/blob/17ceab7952fc67674b624183b62eae16ecd12824/core/src/worker/activities.rs#L197

So if we break out of the select statement after adding a poll permit, the next poll request might add an additional permit.

https://github.com/temporalio/sdk-core/blob/17ceab7952fc67674b624183b62eae16ecd12824/core/src/worker/activities.rs#L200

My point is that this mechanism needs a better design and the fact that we sometimes get more than one poller is not the way it should work.

I hope I'm not terribly misinterpreting the code.

Reply:
Yes it's totally true that it's nonobvious why it works, but essentially it works because a permit is added to the polls requested sempahore every time lang calls, but it is not used every time, because the poll may return early because of buffered work.

So, yeah, it should be changed for sure, but let's stop saying it only ever polls one at a time b/c that's not accurate either.